jina/executors/indexers/vector.py

```diff
@@ -123,78 +130,92 @@ def build_advanced_index(self, vecs: 'np.ndarray'):
         raise NotImplementedError
 
     def _load_gzip(self, abspath: str) -> Optional['np.ndarray']:
-        self.logger.info(f'loading index from {abspath}...')
-        if not path.exists(abspath):
-            self.logger.warning('numpy data not found: {}'.format(abspath))
-            return None
-        result = None
         try:
-            if self.num_dim and self.dtype:
-                with gzip.open(abspath, 'rb') as fp:
-                    result = np.frombuffer(fp.read(), dtype=self.dtype).reshape([-1, self.num_dim])
+            self.logger.info(f'loading index from {abspath}...')
+            with gzip.open(abspath, 'rb') as fp:
+                return np.frombuffer(fp.read(), dtype=self.dtype).reshape([-1, self.num_dim])
         except EOFError:
             self.logger.error(
                 f'{abspath} is broken/incomplete, perhaps forgot to ".close()" in the last usage?')
-        return result
 
     @cached_property
     def raw_ndarray(self) -> Optional['np.ndarray']:
-        vecs = self._load_gzip(self.index_abspath)
-        if vecs is None:
-            return None
+        if not (path.exists(self.index_abspath) or self.num_dim or self.dtype):
+            return
 
-        if self.key_bytes and self.key_dtype:
-            self.int2ext_key = np.frombuffer(self.key_bytes, dtype=self.key_dtype)
+        if self.compress_level > 0:
+            return self._load_gzip(self.index_abspath)
+        elif self.size is not None and os.stat(self.index_abspath).st_size:
```

Comment:
an extra else: raising an Exception would help to tackle the problem if ever found. Maybe u run it without the proper previous `index` flow or something?

Reply:
will just return `None`, `get_query_handler` will handle `None`